JSDemos/Demos/DataGrid/RecordPaging/Angular/app/app.component.html

```diff
@@ -1,17 +1,55 @@
 <dx-data-grid 
-    id="gridContainer"
-    [dataSource]="dataSource"
-    [showBorders]="true">    
-    <dxo-paging [pageSize]="10"></dxo-paging>
+    id="gridContainer" 
+    [dataSource]="dataSource" 
+    [showBorders]="true">
+    <dxo-scrolling 
+        rowRenderingMode="virtual">
+    </dxo-scrolling>
+    <dxo-paging 
+        [pageSize]="10">
+    </dxo-paging>
     <dxo-pager 
-        [showPageSizeSelector]="true"
-        [allowedPageSizes]="[5, 10, 20]"
-        [showInfo]="true">
+        [visible]="true" 
+        [allowedPageSizes]="[5, 10, 'all']" 
```

Comment:
I think it is better not to use inline objects directly in markup. Move the array to the component option. For instance:
```html
[allowedPageSizes]="pageSizes"
```
```ts
  readonly pageSizes = [5, 10, 'all'];
```